### PR TITLE
Adds some basic READMEs to some of the test images

### PR DESCRIPTION
**Pull request by johnSchnake**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Clarifies the use/purpose of some of the test images for easier
discovery/use in the future.

**Which issue(s) this PR fixes**:
Related to #76342

**Special notes for your reviewer**:
Very open to rewording/clarifying any of these; they were just first iterations of my own description/understanding of each.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/sig testing
cc @bclau

### CODE REVIEW

File: test/images/ipc-utils/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# ipc-utils
+
+This container will be used to exercise the HostIPC functionality in
+e2e-node tests.
+
+The version of `ipcs` shipped in busybox performs operations that get
+blocked by SELinux on hosts where it is enabled. The version of `ipcs`
+in util-linux does not perform those operations, rather it checks
+whether the /proc files it needs are available and proceeds to reading
```
Comment (bclau) [inline test/images/ipc-utils/README.md]: proceeds to read* ?

File: test/images/metadata-concealment/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# metadata-concealment
+
+Runs various checks against the GCE metadata API. If the checks pass the job is run to completion. It fails and stops at the first error encountered.
```
Comment (bclau) [inline test/images/metadata-concealment/README.md]: s/If the checks pass/If the checks pass,

Added an extra comma, so it's easier to understand.

File: test/images/nonewprivs/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# nonewprivs
+
+A simple go app that prints the UID of the process running to test security context features.
```
[the remark this answers is not in the document]
Comment (johnSchnake) [reply test/images/nonewprivs/README.md]: I hadn't seen this image before either; rolling it into the agnhost image would be trivial was my thought too but would be a separate issue and so I went ahead and documented it here.

File: test/images/ipc-utils/README.md (hunk 1 above)
Comment (johnSchnake) [reply test/images/ipc-utils/README.md]: 👍 I just copy/pasted from the original PR that added this image (just for context)

**Comment by mkumatag** [comment]
I'm just piggypacking on @bclau approval and merging this PR.
/approve

**Pull request merged**